### ISSUE TITLE
HomeConnect Node does not connect anymore

### ISSUE THREAD

**Issue by afrixx** [alexkn/node-red-contrib-homeconnect#99]
Hello,

my home connect event node cannot connect anymore to the API Server.
The credentials did not change.

When I click at the "Start authorization" button in Node Red, another tab opens with 403 Error (https://api.home-connect.com/...).
Did the API change? 
Does anybody have the same problem?

Thank you!
Andreas

**Comment by drbacke** [comment]
Same Problem here. Error 403

**Comment by TheDooney78** [comment]
Same problem for me. I always get error 403

**Comment by pnuding** [comment]
Same issue - checked I can sign in to the Home Connect developer page normally, but the node-red authorization flow now fails as described above. 
Has anyone managed to report the issue to Home Connect? I've tried using the contact form but get a message that "Antibot verification failed" so it won't let me submit

**Comment by afrixx** [comment]
It seems to be a temporary issue. In between it worked again, but also failed again.

I wrote a message to the support (the second sending attempt worked without Antibot verification error), but did not get any response.

**Comment by michelscholte** [comment]
See my post "Error 403 since last week #98"which I closed a few weeks ago.. After making an issue I got an email back from the developer team: 

Hello Michel.

Sorry for inconvenience, we are aware of this issue, and we are fixing it.

Please try again in a day or two and let me know if there will be still a problem.

Best Regards,

Your Home Connect Developer Team

https://developer.home-connect.com/

Last week, everything was working smoothly, but starting today, I'm getting the same 403 error again.
Hopefully, this is just a temporary issue. I'll reach out via email once more and keep you updated.

**Comment by pnuding** [comment]
Still broken 😢

**Comment by drbacke** [comment]
Same Here

**Comment by drbacke** [comment]
For authorization the below link -> 403 Forbidden:
URL/Code Example:
https://api.home-connect.com/security/oauth/authorize?client_id=XY&response_type=code&redirect_uri=XY
For me it doesn't work all the time

**Comment by michelscholte** [comment]
I haven't received any further responses from support, but the system has been working for me now for almost two weeks (just after my last post). The issue now only occurs when Node-RED is restarted. To resolve it, I do these steps:

Check the "Use Simulation" checkbox.
Sign in and accept the prompts.
Click Update and Deploy in Node-RED.
Uncheck the "Use Simulation" box.
Sign in and accept again.
Click Update and Deploy in Node-RED once more.
After these steps, the connection is re-established (in my case).

![Knipsel](https://github.com/user-attachments/assets/37b4d223-721e-4bac-a417-681ee55ab45d).

You can also try using the computer-name instead of the IP address in the Callback URL, both in Node-RED and on the developer webpage. I had some trouble with this also.
![Knipsel](https://github.com/user-attachments/assets/080bdab3-90d2-4b11-ba8f-c5282ce16605)

**Comment by pnuding** [comment]
Thanks michelscholte! Neither the checking/unchecking nor the changing to the computer name (i had the IP address before) solved it. But i found that deleting my existing homeconnect application and setting up a new one did work. I've now used the computer name, haven't tried if it would have accepted the IP address

**Comment by drbacke** [comment]
Yes that worked. The Hostname was Not accepted, an IP results in 403.
Server as hostname is also prohibited😵‍💫

**Comment by 8xiom** [comment]
Stumbled across the same problem. Thanks to @michelscholte for the hint. Unfortunately, this only works once in my case. As soon as I reconfigure the node (which is necessary because I have still to figure out the correct way to receive the data from my HC appliance, a Siemens coffee machine) the authorization is lost. I have to do the steps over and over again, only works once.

**Comment by FireWizard52** [comment]
@8xiom wrote:

> Unfortunately, this only works once in my case. As soon as I reconfigure the node (which is necessary because I have still to figure out the correct way to receive the data from my HC appliance, a Siemens coffee machine) the authorization is lost. I have to do the steps over and over again, only works once.

I have the same problem. As soon as I have created a new application, it works once.
If I reconfigure the node, I receive: `"auth not ready"`.

**Comment by belveder79** [comment]
Does not work for me either - Simulator works but regular does not. 

I just came across a different [concern](https://github.com/alexkn/node-red-contrib-homeconnect/issues/100) and wanted to hear your thoughts about that...

**Comment by FireWizard52** [comment]
Hello @belveder79 

> I just came across a different https://github.com/alexkn/node-red-contrib-homeconnect/issues/100 and wanted to hear your thoughts about that...

See my answer in that other topic.

**Comment by FireWizard52** [comment]
I think I found something.

I had the same issue if I use http://hostname:1880/homeconnect/auth/callback. It works once.
It connects and you will get your appliances, but as soon as you change the node, it disconnect and will not connect again.

I changed the callback URL into http://hostname:1880//oauth2/auth/callback.
I found this information at: https://github.com/pckhib/node-red-contrib-homeconnect?tab=readme-ov-file

It started to work immediately for me. However if I add a new Home Connect Request node it disconnects again. Playing a little bit with the authorization and switching the callback URL shows again "ready" for all "Request" nodes and "connected" for the "Event". So I think the authorization node is somewhat buggy.

I have never tried it with the IP address, so I cannot confirm, that this shows the same behavior.

I suggest to other users with the same problem to test that other callback URL and report back.

Best Regards

**Comment by ringostarlet** [comment]
This workaround doesn't work for me, the authorization is rejected, also in simulator mode. With the regular callback I can register as simulator but not for real. Upon changing the node the authorization is gone again. Any ideas?

**Comment by Marooned-MB** [comment]
Is this node working for anyone?
When used [v] Simulation, then I get proper auth. When unchecked, I get 403.

For Simulation checked in, I get `Connected` status but error `"TypeError: Cannot read property 'get_home_appliances' of undefined"`
For Simulation checked out, I get `Not connected` status and error
```
refreshTokens failed: {
  "error": "invalid_grant",
  "error_description": "invalid refresh_token"
}
```

**Comment by FireWizard52** [comment]
Hello @Marooned-MB,

You asked:

> Is this node working for anyone?

Yes, it works for me, but I have no practical use for it at the moment, so I really don't use it frequently

<img width="1124" height="434" alt="Image" src="https://github.com/user-attachments/assets/14740b8c-f70a-48f2-a23e-37696bc68525" />

I do not have similar issues.

Regards

**Comment by n1ghty** [comment]
I had the same issue. Fixed it by manually removing the complete"redirect_uri=http...." parameter from the url when I got the 403 error.
The login is displayed and the redirect also works because it default to the setting in the application config in the developer portal.

So it can be fixed by not sending the redirect_uri at all.

This should work:
Remove line 119 in lib/ApiService.js
```javascript
        '&redirect_uri=' + encodeURIComponent(callback_url) +
```

**Comment by afrixx** [comment]
> I had the same issue. Fixed it by manually removing the complete"redirect_uri=http...." parameter from the url when I got the 403 error. The login is displayed and the redirect also works because it default to the setting in the application config in the developer portal.
> 
> So it can be fixed by not sending the redirect_uri at all.
> 
> This should work: Remove line 119 in lib/ApiService.js
> 
>         '&redirect_uri=' + encodeURIComponent(callback_url) +

This worked for me!
It seems that you have to re-authenticate after a certain time. When you do this via Node-Red UI, you will probably get the 403 error. When removing the redirect parameter, you can log in and grant the permissions.
After that, save the changes and do a new "deploy" (maybe you have to move a node somewhere before).
Then the re-connect worked for me.

**Comment by Marooned-MB** [comment]
I can confirm that removing `redirect_uri=http....` params from the url when presented to 403 page redirects me to a login page.
After login however I get redirected to `https://my.home-assistant.io/redirect/oauth/?code=...` page with alert popup `Invalid parameters given.`.
After closing that alert I get redirected to https://my.home-assistant.io/create-link/?redirect=oauth with `Create a link` form that I don't understand what is it for.

On my HA I can see label `authorizing...` next to `Start authorization` button.

One level above it requires from me HAID field that I have no idea where to find but I assume it's a separate topic.
